src/components/Item.js

```diff
@@ -0,0 +1,72 @@
+import React, { useEffect, useState } from 'react';
+import '../components/AddItemForm.css';
+import { updateDoc, serverTimestamp, doc } from 'firebase/firestore';
+import { db } from '../lib/firebase';
+import './Item.css';
+
+function Item({ item, i }) {
+  const [checked, setChecked] = useState(false);
+  const [daysSincePurchased, setDaysSincePurchased] = useState();
+  const userToken = window.localStorage.getItem('userToken');
+
+  useEffect(() => {
+    if (item.lastPurchased) {
+      const lastPurchasedSeconds = item.lastPurchased.seconds;
+      const dateNowSeconds = Date.now() / 1000;
+      const differenceInSeconds = dateNowSeconds - lastPurchasedSeconds;
+
+      setDaysSincePurchased(differenceInSeconds / 86400);
+      if (differenceInSeconds / 86400 < 1) {
```

Comment:
Suggestion: You could use the `daysSincePurchased` state variable here if you move it into the below `useEffect` that already has that as a dependency.

It keeps a level of readability and consistency that way, in that the `useEffect` will monitor the state of `daysSincePurchased` and then check if it is greater than or less than `1` and set `checked` accordingly.

Reply:
@abowler2 Thanks for this suggestion! It makes it that section of code more understandable.